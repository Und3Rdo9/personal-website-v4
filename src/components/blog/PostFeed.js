import React from 'react';
import PropTypes from 'prop-types';
import PostFeedItem from './PostFeedItem';

const PostFeed = ({ postsData }) => {
  const renderPosts = (data) => {
    if (!data.length) return null;
    return data.map((postData) => {
      return <PostFeedItem post={postData} key={postData.slug} />
    })
  }
  return (
    <div className="post-feed">
      {renderPosts(postsData)}
    </div>
  )
};

PostFeed.propTypes = {
  postsData: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default PostFeed;
