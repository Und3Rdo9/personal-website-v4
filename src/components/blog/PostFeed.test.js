import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import PostFeed from './PostFeed';

configure({ adapter : new Adapter() });

describe('<PostFeed />', () => {

  const testProps = {
    postsData : [
    {
      title: 'testTitle',
      summary: 'testSummary',
      slug: 'test-slug',
      coverImage: {
        url: 'http://timbakkum.net'
      },
      dateAndTime: '21380921389'
    },
    {
      title: 'testTitle2',
      summary: 'testSummary2',
      slug: 'test-slug2',
      coverImage: {
        url: 'http://timbakkum2.net'
      },
      dateAndTime: '29991002001'
    }]
  };

  const wrapper = shallow(
    <PostFeed {...testProps} />
  );

  it('renders correctly ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays the correct amount of PostFeedItem components', () => {
    expect(wrapper.find('PostFeedItem').length).toBe(2);
  });
});
