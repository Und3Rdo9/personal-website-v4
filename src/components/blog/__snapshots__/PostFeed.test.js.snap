// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostFeed /> renders correctly  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostFeed
    postsData={
        Array [
            Object {
              "coverImage": Object {
                "url": "http://timbakkum.net",
              },
              "dateAndTime": "21380921389",
              "slug": "test-slug",
              "summary": "testSummary",
              "title": "testTitle",
            },
            Object {
              "coverImage": Object {
                "url": "http://timbakkum2.net",
              },
              "dateAndTime": "29991002001",
              "slug": "test-slug2",
              "summary": "testSummary2",
              "title": "testTitle2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PostFeedItem
          post={
                    Object {
                              "coverImage": Object {
                                "url": "http://timbakkum.net",
                              },
                              "dateAndTime": "21380921389",
                              "slug": "test-slug",
                              "summary": "testSummary",
                              "title": "testTitle",
                            }
          }
/>,
        <PostFeedItem
          post={
                    Object {
                              "coverImage": Object {
                                "url": "http://timbakkum2.net",
                              },
                              "dateAndTime": "29991002001",
                              "slug": "test-slug2",
                              "summary": "testSummary2",
                              "title": "testTitle2",
                            }
          }
/>,
      ],
      "className": "post-feed",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "test-slug",
        "nodeType": "function",
        "props": Object {
          "post": Object {
            "coverImage": Object {
              "url": "http://timbakkum.net",
            },
            "dateAndTime": "21380921389",
            "slug": "test-slug",
            "summary": "testSummary",
            "title": "testTitle",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "test-slug2",
        "nodeType": "function",
        "props": Object {
          "post": Object {
            "coverImage": Object {
              "url": "http://timbakkum2.net",
            },
            "dateAndTime": "29991002001",
            "slug": "test-slug2",
            "summary": "testSummary2",
            "title": "testTitle2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PostFeedItem
            post={
                        Object {
                                    "coverImage": Object {
                                      "url": "http://timbakkum.net",
                                    },
                                    "dateAndTime": "21380921389",
                                    "slug": "test-slug",
                                    "summary": "testSummary",
                                    "title": "testTitle",
                                  }
            }
/>,
          <PostFeedItem
            post={
                        Object {
                                    "coverImage": Object {
                                      "url": "http://timbakkum2.net",
                                    },
                                    "dateAndTime": "29991002001",
                                    "slug": "test-slug2",
                                    "summary": "testSummary2",
                                    "title": "testTitle2",
                                  }
            }
/>,
        ],
        "className": "post-feed",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "test-slug",
          "nodeType": "function",
          "props": Object {
            "post": Object {
              "coverImage": Object {
                "url": "http://timbakkum.net",
              },
              "dateAndTime": "21380921389",
              "slug": "test-slug",
              "summary": "testSummary",
              "title": "testTitle",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "test-slug2",
          "nodeType": "function",
          "props": Object {
            "post": Object {
              "coverImage": Object {
                "url": "http://timbakkum2.net",
              },
              "dateAndTime": "29991002001",
              "slug": "test-slug2",
              "summary": "testSummary2",
              "title": "testTitle2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
